### ISSUE TITLE
hashcat style permutation system

### ISSUE THREAD

**Issue by aoighost** [vanhauser-thc/thc-hydra#558]
It would be useful if we could take a partial password or a known wordlist and permute based off it in brute forcing similar to how hashcat works.

**Comment by vanhauser-thc** [comment]
yes that would be nice, and I am open to receive a pull request for that :)

as a workaround you can also use john/hashcat to create a wordlist from them and then pass that to hydra.

**Comment by Nj0n** [comment]
the problem is to generate a password list is very slow since it is working only on cpu.
same hydra while brute forcing ? cause even rocket.txt takes ages to crack.
any ideas?

**Comment by vanhauser-thc** [comment]
has nothing to do with cpu vs gpu. network latency and server resources to a client are the bottleneck

**Comment by Rayanmanini** [comment]
Nikmon

**Comment by h4ck3r-04** [comment]
Since Hydra offers fewer options than Hashcat, I'm considering the idea of adapting password generation scripts from Hashcat to integrate into Hydra. Additionally, we could explore the possibility of incorporating an option for utilizing a rules configuration file to manage multiple rules.
